src/game/AI/ScriptDevAI/scripts/outland/hellfire_citadel/shattered_halls/mob_shattered_hand_legionnaire.cpp

```diff
@@ -146,21 +151,35 @@ struct mob_shattered_hand_legionnaireAI : public ScriptedAI
                 DoScriptText(EMOTE_ENRAGE, m_creature);
             }
 
-            // Have to reasearch the cooldown for FelOrc spawn
-            if (m_uiMinionSpawnTimmer < uiDiff)
-            {                
+            // Legionnaire from group 5 "awakes" one from the nearby sleeping creatures
+            if (legionnaireGuid == 5)
+            {
 
-                if (Creature* felorc = m_creature->SummonCreature(MOB_FEL_ORC, FelOrcSpawnCoords[legionnaireGuid][0], FelOrcSpawnCoords[legionnaireGuid][1], FelOrcSpawnCoords[legionnaireGuid][2], FelOrcSpawnCoords[legionnaireGuid][3], TEMPSPAWN_TIMED_OOC_OR_DEAD_DESPAWN, 100000, true, true))
+                if (Creature* sleeping = GetClosestCreatureWithGuid(m_creature, aSleepingReinfGuid[urand(0, 3)], 150.0f))
```

Comment:
This is an intersting part for me.
On last legionnaire group before first boss, if a legionnaire group member dies, no fel orc spawn. instead one of the 4 sleeping creatures in this room gets awaken and joining the fight.
My first thought was instead of doing this guid based with new GetClosestCreatureWithGuid function, 
adding a function like DoFindHavingBuff, checking for all creatures that have the sleeping aura.

but i prefered this way